rest/src/main/java/com/confighub/api/repository/client/v1/APIPull.java

```diff
@@ -163,6 +159,33 @@ public Response get(@PathParam("account") String account,
         }
     }
 
+    private JsonObject getConfiguration(Gson gson,
+                                        String contextString,
+                                        boolean includeComments,
+                                        boolean noFiles,
+                                        boolean noProperties,
+                                        boolean includeContext)
+    {
+        JsonObject json = null;
+        if (!repository.isCachingEnabled() || !cache.containsKey(context))
+        {
+            json = getConfiguration(repository, context, resolved, passwords, new JsonObject(), noFiles, noProperties,
+                    includeComments, includeContext, gson);
+        }
+
+        if (repository.isCachingEnabled())
```

Comment:
i generally like to reverse the order of these ifs so that you can return the json from the cache and not do double hash lookups.

if(repository.isCachingEnabled()) // is this going in memory or to the db each time?
{
  json = cache.getOrDefault(context, null);
  if(json != null)
  {
    addJsonHeader(json, repository, contextString, context);
    return json;
  }
}
json = getConfiguration(...);
if(repository.isCachingEnabled()) // may want to move this above if it hits the db each time
{
  cache.putIfAbsent(context, json);
}
addJsonHeader(...);
return json;

Now the question is do we need to do some concurrency checking here?  I see alot of re-going back to the cache (i assume) in case it gets written by another thread.  To me if java has a GetOrAdd functionality that would be awesome.